### ISSUE TITLE
Add helpers for snapshots

### ISSUE THREAD

**Issue by nventuro** [OpenZeppelin/openzeppelin-test-helpers#101]
@jamesmorgan suggested on the Telegram channel adding support for `evm_snapshot` and `evm_revert`. Given that their semantics are a bit strange, and that a loot of speed can be gained with proper usage of snapshots, I think this is a great idea.

I'm not sure on the API, we could definitely simply provide simple functions to call these, but it might be possible to achieve a more ergonomic solution. @jamesmorgan could you share more about your use case to see if we can tailor our solution to it? Thanks!

**Comment by jamesmorgan** [comment]
Hi guys, ye we've have been using the following methods with some success on a project which uses `Builder` but our main dev tool kit is OZ. When I recently looked in the helpers I noticed it wasn't their and seems like a obvious improvement.  

In general we have been using snapshots to speed up tests which have lots of nuanced scenarios which require much setup, fundamentally speeding up the suite. 

```
async function snapshot () {
  return ethereum.send('evm_snapshot', [])
}

async function restore (snapshotId) {
  return ethereum.send('evm_revert', [snapshotId])
}
```

Anyway, would love to see it added. I will try get round to adding a PR with come basic utility methods in this week.
